docs/content/basics/testing_theory.md

```diff
@@ -1,57 +1,163 @@
-<br/>But don't forget that this is essentially a question of the acceptable quality of a particular product.
+Apart from their granularity, there are two interesting means to classify tests, as pictured below
+
+![Group 5.png](https://cdn.hashnode.com/res/hashnode/image/upload/v1637010289284/lS74gW9vvt.png)
+
+!!! Disclaimer
+
+      Important to note that these classifications are not mutually exclusive. One can write, for instance
+
+      1. Ui tests that run on the JVM and only test one view (i.e. unit test)
+      2. Non-Ui tests that run on a device (i.e. instrumented) and test several components (i.e. integration test)
+
+## Classifying tests by the environment where they can run
+Depending on the environment the tests can run, we get the following distribution
+
+1. **JVM (Java Virtual Machine) tests**: can run without the need of an emulator or physical device.
+
+2. **Instrumentation tests**: those that require an emulator or physical device. These are tests that
+   contain Android-specific code, like Views. Such code requires a DVM (Dalvik Virtual Machine) or Android Runtime (ART) since 5.0 to be executed, and therefore it cannot run on the JVM but on a device.
+
+3. **Shared tests**: These are tests that are written once, but can run either as JVM or as Instrumentation tests. The principle is simple: *Write once, run everywhere*.
+   These tests contain Android-specific code. Such code would need a device to run on. In order to enable that, when running on the JVM, that code is replaced by mocks under the hood. We'll go deeper into this later on.
+
+## Classifying tests by their target
+Depending on the target of our tests, they are split into the following categories
+
+1. **UI tests**: focus on testing the WHAT (interaction/navigation); *WHAT is displayed* when interacting with the screen elements.
+2. **Screenshot/Snapshot tests**:  focus on testing the HOW (visuals): *HOW a view is displayed* under a given state or configuration.
+3. **Non-UI tests**: focus on testing non-ui related code like *BUSINESS LOGIC* or *DATABASE MIGRATIONS* among others
+
+### Should I write Snapshot tests or Ui Tests?
+It might be confusing to understand when to write Ui test rather than Snapshot tests and vice versa. They do not replace each other. Their focus is different as previously mentioned. So let's imagine the following screen, which is a RecyclerView
+
+![gif300Faster.gif](https://cdn.hashnode.com/res/hashnode/image/upload/v1631992848600/ajg2fTGRG.gif)
+
+A *Ui test would verify*, e.g. that after deleting a row in the recycler View, that row is not displayed anymore. It would test *WHAT is displayed after interacting with the view*.
+
+On the other hand, a *snapshot test would verify HOW that row is displayed* under numerous states and configurations: e.g. dark/light mode, LTR/RTL languages, different font sizes, wide/narrow screens...
+
```

Comment:
Here I would gladly add an image with some of those configs, e.g. different font size?